tests/integration/controller/kyma/helper_test.go

```diff
@@ -71,9 +63,14 @@ func RegisterDefaultLifecycleForKymaWithoutTemplate(kyma *v1beta2.Kyma) {
 func DeleteModuleTemplates(ctx context.Context, kcpClient client.Client, kyma *v1beta2.Kyma) {
 	for _, module := range kyma.Spec.Modules {
 		template := builder.NewModuleTemplateBuilder().
+			WithName(createModuleTemplateName(module)).
 			WithModuleName(module.Name).
 			WithChannel(module.Channel).
 			WithOCM(compdescv2.SchemaVersion).Build()
+		Eventually(RegularModuleTemplateExists, Timeout, Interval).
```

Comment:
I think we don't need this check, it can happen some integration test testing moduletemplate deletion, this `AfterAll` step is just clean up resources, it's valid case that some resources already gone.